File: new/python/valid_anagram_sorting.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # if they don't have same length they cannot be a anagram
        # else, sort them and check if they are equal
        if len(s) != len(t):
            return False

        return sorted(s) == sorted(t)
        # time: O(nlogn + mlogm) = O(nlogn) because both have same length
        # space: depending on sorting technique: O(1) or O(n), in python O(n) sorting
        # always returns a new list

File: new/cpp/valid_anagram_sorting.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        // if they don't have same length return false
        // sort them and check if they are equal
        if (s.size() != t.size()) {
            return false;
        }

        sort(s.begin(), s.end());
        sort(t.begin(), t.end());

        return s == t; // comparison takes O(n) but sorting dominates so O(nlogn)
        // time: O(nlogn) because sorting in c++ is nlogn
        // space: O(1) or O(n) depending on sorting type
    }
};
